frappe/utils/background_jobs.py

```diff
@@ -186,6 +187,9 @@ def execute_job(site, method, event, job_name, kwargs, user=None, is_async=True,
 	"""Executes job in a worker, performs commit/rollback and logs if there is any error"""
 	retval = None
 
+	actual_func_name = kwargs.get("job_type") if "run_scheduled_job" in method else method
```

Comment:
```suggestion
	actual_func_name = kwargs.get("job_type") if "run_scheduled_job" in method_name else method_name
```

Can move lines 201-205 above and do this, should fix tests

Reply:
Yep. It's the function objects causing failures.